### ISSUE TITLE
Searches over multiple tag services at once involving different queries per service

### ISSUE THREAD

**Issue by Zweibach** [hydrusnetwork/hydrus#1499]
The idea would be to make searches over two or possible more services to give us files that perhaps have a tag on one but not the other, or the same tag on both services. For example if you want to compare character tags on PTR vs. local tags to see if they're correct.

The current workaround is basically to do the first search, copy all the hashes, do a new search with the hashes. Which should give you the desired result but is pretty cumbersome.

```
in:my tags has tag[samus aran]
in:public tag repository has tag[samus aran]
``` 

Would only match files that are tagged with `samus aran` in both of the services.

```
in:my tags has tag[samus aran]
in:public tag repository does not have tag[samus aran]
```

Would match files that are tagged with `samus aran` in my files, but not on PTR.

```
in:my tags does not have tag[samus aran]
in:public tag repository does not have tag[samus aran]
```

Would only give you files that are not tagged with `samus aran` on both services. You would probably need a third query of just `samus aran` to get any files that are tagged in just one service. Would probably depend on how the UI for this looks like.

### UI
For how it would look you could have either something like the duplicates filter where you have two boxes to input your queries with some match options, or perhaps an option to edit a query to match on selected services.
```
samus aran
= my tags AND ptr
```

This should ignore any files tagged in, for example, the downloaded tags service that ships by default.

```
samus aran
= my tags
-samus aran
= ptr
```
As a very rough example that I think matches the first two. Depending on how it works you might even be able to work in some OR logic there as well.

tl;dr I want to be able to do multiple searches with tags on different services at once.
